src/main/kotlin/com.bugsnag.android.gradle/BugsnagMultiPartUploadRequest.kt

```diff
@@ -101,11 +109,60 @@ class BugsnagMultiPartUploadRequest {
      *
      * @return the retry count
      */
-    private fun getRetryCount(bugsnag: BugsnagPluginExtension): Int {
-        return if (bugsnag.retryCount >= MAX_RETRY_COUNT) MAX_RETRY_COUNT else bugsnag.retryCount
+    private fun getRetryCount(): Int {
+        return if (retryCount >= MAX_RETRY_COUNT) MAX_RETRY_COUNT else retryCount
     }
 
     companion object {
         const val MAX_RETRY_COUNT = 5
+
+        internal fun <T> from(
+            task: T
+        ): BugsnagMultiPartUploadRequest where T : DefaultTask, T: BugsnagFileUploadTask {
+            return BugsnagMultiPartUploadRequest(
+                task.logger,
+                failOnUploadError = task.failOnUploadError.get(),
+                overwrite = task.overwrite.get(),
+                endpoint = task.endpoint.get(),
+                retryCount = task.retryCount.get(),
+                timeoutDuration = Duration.ofMillis(task.timeoutMillis.get())
+            )
+        }
+
+        private fun createService(timeoutDuration: Duration): BugsnagService {
+            return createService(OkHttpClient.Builder()
+                .connectTimeout(timeoutDuration)
+                .callTimeout(timeoutDuration)
+                .build())
+        }
+
+        internal fun createService(client: OkHttpClient): BugsnagService {
+            return Retrofit.Builder()
+                .baseUrl("https://example.com") // Not actually used
```

Comment:
would it be possible to default this to the value of `bugsnag.endpoint`, just on the off chance that a future refactor does alter this so that the baseUrl is used?

Reply:
Another, possibly more idiomatic way, would be to use this and then just make the path for the endpoint be "/"